sdk/resourcemanager/azure-resourcemanager-sql/src/main/java/com/azure/resourcemanager/sql/models/DatabaseSku.java

```diff
@@ -0,0 +1,344 @@
+    public static final DatabaseSku HYPERSCALE_HS_GEN5_12 = new DatabaseSku("HS_Gen5", "Hyperscale", "Gen5", 12, null);
+    /** Hyperscale Edition with HS_Gen5_14 sku. */
+    public static final DatabaseSku HYPERSCALE_HS_GEN5_14 = new DatabaseSku("HS_Gen5", "Hyperscale", "Gen5", 14, null);
+    /** Hyperscale Edition with HS_Gen5_16 sku. */
+    public static final DatabaseSku HYPERSCALE_HS_GEN5_16 = new DatabaseSku("HS_Gen5", "Hyperscale", "Gen5", 16, null);
+    /** Hyperscale Edition with HS_Gen5_18 sku. */
+    public static final DatabaseSku HYPERSCALE_HS_GEN5_18 = new DatabaseSku("HS_Gen5", "Hyperscale", "Gen5", 18, null);
+    /** Hyperscale Edition with HS_Gen5_20 sku. */
+    public static final DatabaseSku HYPERSCALE_HS_GEN5_20 = new DatabaseSku("HS_Gen5", "Hyperscale", "Gen5", 20, null);
+    /** Hyperscale Edition with HS_Gen5_24 sku. */
+    public static final DatabaseSku HYPERSCALE_HS_GEN5_24 = new DatabaseSku("HS_Gen5", "Hyperscale", "Gen5", 24, null);
+    /** Hyperscale Edition with HS_Gen5_32 sku. */
+    public static final DatabaseSku HYPERSCALE_HS_GEN5_32 = new DatabaseSku("HS_Gen5", "Hyperscale", "Gen5", 32, null);
+    /** Hyperscale Edition with HS_Gen5_40 sku. */
+    public static final DatabaseSku HYPERSCALE_HS_GEN5_40 = new DatabaseSku("HS_Gen5", "Hyperscale", "Gen5", 40, null);
+    /** Hyperscale Edition with HS_Gen5_80 sku. */
+    public static final DatabaseSku HYPERSCALE_HS_GEN5_80 = new DatabaseSku("HS_Gen5", "Hyperscale", "Gen5", 80, null);
+
+    private final Sku sku;
+
+    /**
+     * Creates a database sku.
+     *
+     * @param name the sku name
+     * @param tier the sku tier
+     * @param family the sku family
+     * @param capacity the sku capacity
+     * @param size the sku size
+     */
+    public DatabaseSku(String name, String tier, String family, Integer capacity, String size) {
+        this.sku = new Sku().withName(name).withTier(tier).withFamily(family).withCapacity(capacity).withSize(size);
+        VALUES.putIfAbsent(toString(), this);
+    }
+
+    /**
+     * Parses a sku to a DatabaseSku instance.
+     *
+     * @param sku the sku.
+     * @return the DatabaseSku object, or null if sku is null.
+     */
+    public static DatabaseSku fromSku(Sku sku) {
+        if (sku == null) {
+            return null;
+        }
+        return new DatabaseSku(sku.name(), sku.tier(), sku.family(), sku.capacity(), sku.size());
+    }
+
+    /**
+     * Lists the pre-defined database sku.
+     *
+     * @return immutable collection of the pre-defined database sku
+     */
+    public static Collection<DatabaseSku> getAll() {
+        return VALUES.values();
+    }
+
+    @Override
+    public String toString() {
+        return String.format("%s/%s/%s/%d/%s", sku.name(), sku.tier(), sku.family(), sku.capacity(), sku.size());
+    }
+
+    /** @return the underneath sku description */
+    @JsonValue
+    public Sku toSku() {
+        return this.sku;
```

Comment:
This might better return a new `Sku` object.

Else user might do `toSku().withTier(...)` to get the underlying `Sku` modified.

Reply:
done